src/dbnode/storage/index/block.go

```diff
@@ -772,23 +774,50 @@ func (b *block) Query(
 	}
 
 	size := results.Size()
-	limitedResults := false
+	batch := b.docsPool.Get()
+	batchSize := cap(batch)
+	if batchSize == 0 {
+		batchSize = defaultQueryDocsBatchSize
+	}
+	batchOpts := AddDocumentsBatchResultsOptions{
+		AllowPartialUpdates: true,
+		LimitSize:           opts.Limit,
+	}
 	iterCloser := safeCloser{closable: iter}
 	execCloser := safeCloser{closable: exec}
 
 	defer func() {
+		b.docsPool.Put(batch)
```

Comment:
Do we need to reset the batch here?

Reply:
No thankfully the Put method on the array will zero out each elem and also resize it to zero.